tests/jobs/test_backfill_job.py

```diff
@@ -1538,3 +1538,25 @@ def test_mapped_dag(self, dag_id):
             executor=SequentialExecutor(),
```

Comment:
Ah reading the new test case makes me remember we actually needed this in the first place for this particular reason! Maybe we can remove this now or parametrize the executor to merge the two test functions?

Reply:
Have done that.